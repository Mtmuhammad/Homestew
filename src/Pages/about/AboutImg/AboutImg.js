import React from "react";
import "./AboutImg.scss";

const AboutImg = ({ img, classes }) => {
  return (
    <div className={`col-6 col-sm-4 ${classes || ""}`}>
      <img
        className="mb-4 about-img img-fluid"
        src={img}
        alt="About"
      />
    </div>
  );
};

export default AboutImg;
