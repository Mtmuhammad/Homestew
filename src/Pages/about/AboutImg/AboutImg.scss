@use "../../../Styles/abstracts/mixins" as *;

.about-img {
   height: 170px;
   border-radius: 5px;

   @include respond(xsphone) {
      height: 150px;
   }
   @include respond(tiny) {
      height: 135px;
   }
}