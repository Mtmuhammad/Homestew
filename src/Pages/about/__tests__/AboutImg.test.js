import React from "react";
import { render, screen } from "@testing-library/react";
import AboutImg from "../AboutImg/AboutImg";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", async () => {
    render(<AboutImg img={"About1.jpg"} />);
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(<AboutImg img={"About1.jpg"} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render an image", () => {
  render(<AboutImg img={"About1.jpg"} />);
  const img = screen.getByAltText(/about/i);
  expect(img).toBeInTheDocument();
  expect(img).toHaveClass("img-fluid", "about-img");
});
