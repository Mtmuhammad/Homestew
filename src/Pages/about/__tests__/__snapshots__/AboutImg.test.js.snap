// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-6 col-sm-4 "
  >
    <img
      alt="About"
      class="mb-4 about-img img-fluid"
      src="About1.jpg"
    />
  </div>
</DocumentFragment>
`;
